mobile/native/src/api_model/position.rs

```diff
@@ -0,0 +1,65 @@
+use crate::api_model::ContractSymbol;
+use crate::api_model::Direction;
+use crate::trade::position::PositionStateTrade;
+use crate::trade::position::PositionTrade;
+use flutter_rust_bridge::frb;
+
+#[frb]
+#[derive(Debug, Clone)]
+pub enum PositionState {
+    /// The position is open
+    ///
+    /// Open in the sense, that there is an active position that is being rolled-over.
+    /// Note that a "closed" position does not exist, but is just removed.
+    /// During the process of getting closed (after creating the counter-order that will wipe out
+    /// the position), the position is in state "Closing".
+    ///
+    /// Transitions:
+    /// Open->Closing
+    Open,
+    /// The position is in the process of being closed
+    ///
+    /// The user has created an order that will wipe out the position.
+    /// Once this order has been filled the "closed" the position is not shown in the user
+    /// interface, so we don't have a "closed" state because no position data will be provided to
+    /// the user interface.
+    Closing,
+}
+
+#[frb]
+#[derive(Debug, Clone)]
+pub struct Position {
+    pub leverage: f64,
+    pub quantity: f64,
+    pub contract_symbol: ContractSymbol,
+    pub direction: Direction,
+    pub average_entry_price: f64,
```

Comment:
What units are these numbers in? I assume unrealised pnl is in sats, but how about the prices here? Are these the price of BTC in the trading pair? If so, does this model still work for quanto CFDs like ETH/USD? I see no obvious reason aside from that there is no BTC side anymore so which side is which becomes arbitrary.

Reply:
> I assume unrealised pnl is in sats

It can also be represented as percentage

> Are these the price of BTC in the trading pair? 

The BTC/USD price to be precise - we could also price our CFDs in e.g. Euro - but that's a smaller market.

> If so, does this model still work for quanto CFDs like ETH/USD?

I am not sure I can perfectly explain the relationship between the price and the margin (in BTC) in a quanto product - but in a nutshell you take bother assets price based on the common price denominator (USD) and the rest is calculation. But that is not really relevant for the `average_entry_price`.

The average entry price would be the ETH/USD price. This average entry price is just an accumulation of prices when the position was opened / updated by orders, so it's an aggregate of the filled order prices that defined the position.

CC @bonomat mayb have a quick read over this just in case, maybe I'm missing something ;)

Reply:
What you said makes sense. 

- the `average_entry_price` is in USD (as long as the contracts are against USD, e.g. ETH/USD, XMR/USD, TESLA/USD). We have an average because there might be multiple trades to make this position.
- `unrealized_pnl` can be in sats or in percent. @da-kami : it might make sense to specify the type in the name, e.g. `unrealized_pnl_sats`. Then it is clear what it is. One can then easily be calculated from the other.